asf_public_discourse_home_decarbonisation/analysis/ASF_awayday_buildhub_eda.py

```diff
@@ -0,0 +1,200 @@
+    <collection_date_time>: data collection date/time in the format YYMMDD (e.g. 231120)
+The figures are saved in the following directory: 'outputs/figures/buildhub/forum_<category>/'
+"""
+# import packages
+import pandas as pd
+import matplotlib.pyplot as plt
+from typing import Optional, Dict, List
+import seaborn as sns
+from wordcloud import WordCloud
+import re
+from collections import Counter
+import nltk
+from nltk import FreqDist
+import argparse
+import os
+from asf_public_discourse_home_decarbonisation import PROJECT_DIR
+from asf_public_discourse_home_decarbonisation.utils.plotting_utils import (
+    set_plotting_styles,
+    finding_path_to_font,
+    NESTA_COLOURS,
+    plot_post_distribution_over_time,
+    plot_users_post_distribution,
+    plot_word_cloud,
+    plot_top_ngrams,
+    plot_post_length_distribution,
+    plot_tag_vertical_bar_chart,
+    plot_tag_horizontal_bar_chart,
+)
+from asf_public_discourse_home_decarbonisation.utils.preprocessing_utils import (
+    preprocess_text,
+    process_ngrams,
+    wordcloud_preprocess_ngrams,
+    update_keyword_frequencies,
+    prepare_keyword_dataframe,
+)
+from asf_public_discourse_home_decarbonisation.getters.bh_getters import (
+    get_bh_category_data,
+)
+from heating_technologies_ruleset import heating_technologies_ruleset_twitter
+
+set_plotting_styles()
+
+font_path_ttf = finding_path_to_font("Averta-Regular")
+
+
+def create_argparser() -> argparse.ArgumentParser:
+    """
+    Creates an argument parser that can receive the following arguments:
+    - category: category or sub-forum (defaults to "119_air_source_heat_pumps_ashp")
+    - collection_date_time: collection date/time (defaults to "231120")
+    Returns:
+        argparse.ArgumentParser: argument parser
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--category",
+        help="Category or sub-forum",
+        default="119_air_source_heat_pumps_ashp",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--collection_date_time",
+        help="Collection date/time",
+        default="231120",
+        type=str,
+    )
+
+    args = parser.parse_args()
+    return args
+
+
+# This is where script is executed.
+if __name__ == "__main__":
+    args = create_argparser()
+    category = args.category
+    collection_date_time = args.collection_date_time
+    BUILDHUB_FIGURES_PATH = os.path.join(
+        PROJECT_DIR, f"outputs/figures/buildhub/forum_{category}/"
+    )
+    # Ensure the output directory exists
+    os.makedirs(BUILDHUB_FIGURES_PATH, exist_ok=True)
+    buildhub_ashp_dataframe = get_bh_category_data(category, collection_date_time)
+    buildhub_ashp_dataframe["date"] = pd.to_datetime(buildhub_ashp_dataframe["date"])
+    post_count = buildhub_ashp_dataframe[
+        buildhub_ashp_dataframe["post_type"] == "Original"
+    ].shape[0]
+    print("post count:")
+    print(post_count)
```

Comment:
In the web scraping scripts we used print because logging doesn't work with Metaflow, but at Nesta and DSP we typically favour logging over printing.

You can do the following at the top of your script
```
import logging
logger = logging.getLogger(__name__)
```

and then here you can do
```suggestion
    logger.info(f"Number of posts: {post_count}")
```

Reply:
Ah yes, this was in DSP conventions! I am very used to using print hehe